packages/fast-check/src/arbitrary/option.ts

```diff
@@ -1,19 +1,22 @@
 import { constant } from './constant';
 import type { Arbitrary } from '../check/arbitrary/definition/Arbitrary';
-import type { _Constraints as FrequencyContraints } from './_internals/FrequencyArbitrary';
+import type { _Constraints as FrequencyConstraints } from './_internals/FrequencyArbitrary';
 import { FrequencyArbitrary } from './_internals/FrequencyArbitrary';
 import type { DepthIdentifier } from './_internals/helpers/DepthContext';
 import type { DepthSize } from './_internals/helpers/MaxLengthFromMinLength';
 import { safeHasOwnProperty } from '../utils/globals';
 
+const safeMathMax = Math.max;
+
 /**
  * Constraints to be applied on {@link option}
  * @remarks Since 2.2.0
  * @public
  */
 export interface OptionConstraints<TNil = null> {
   /**
-   * The probability to build a nil value is of `1 / freq`
+   * The probability to build a nil value is of `1 / freq`.
+   * If `freq` is `0`, then nil is never generated.
```

Comment:
Since 1/0 is commonly not defined but `freq=0` is supported so I thought it's worth explaining that special case.